### PR TITLE
azure: adapt common provider framework

### PR DESCRIPTION
**Pull request by knrt10**
Fixes: #576
Signed-off-by: Kautilya Tripathi <ktripathi@microsoft.com>

### CODE REVIEW

File: cmd/cloud-api-adaptor/main.go [hunk 1]
```diff
@@ -143,8 +116,8 @@ func (cfg *daemonConfig) Setup() (cmd.Starter, error) {
 
 	if hypcfg.HypProvider == "libvirt" {
 		hypervisorServer = registry.NewServer(hypcfg, libvirtcfg, workerNode, daemon.DefaultListenPort)
-	} else if hypcfg.HypProvider == "azure" {
-		hypervisorServer = registry.NewServer(hypcfg, azurecfg, workerNode, daemon.DefaultListenPort)
+	} else if hypcfg.HypProvider == "vsphere" {
+		hypervisorServer = registry.NewServer(hypcfg, vspherecfg, workerNode, daemon.DefaultListenPort)
```
Comment (yoheiueda) [inline cmd/cloud-api-adaptor/main.go:120]: vSphere has been migrated to the new code layout. Could you remove these lines?

File: cmd/cloud-api-adaptor/main.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ type networkConfig struct {
 	VXLANMinID    int
 }
 
-var azurecfg azure.Config
+var vspherecfg vsphere.Config
```
Comment (yoheiueda) [inline cmd/cloud-api-adaptor/main.go:39]: vSphere has been migrated to the new code layout. Could you remove this line?

Comment (knrt10) [reply cmd/cloud-api-adaptor/main.go:39]: Thank you, might have been missed during rebase. Fixed

**Review by yoheiueda — APPROVED** [review]
LGTM

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @knrt10

**Pull request merged**